minishift/README.adoc

```diff
@@ -0,0 +1,217 @@
+[[running-auth-cluster-services-on-openshift]]
+= Running Auth, Cluster services on OpenShift
+:icons:
+:toc: macro
+:toc-title:
+:toclevels: 2
+
+toc::[]
+
+These instructions will help you run your services on OpenShift using MiniShift.
+[[prerequisites]]
+== Prerequisites
+
+link:https://docs.okd.io/latest/minishift/getting-started/installing.html[MiniShift v1.27.0]
+
+link:https://docs.okd.io/latest/cli_reference/get_started_cli.html#installing-the-cli[oc 3.11.0]
+
+link:https://www.linux-kvm.org/page/Downloads[KVM Hypervisor]
+
+[[install-minishift]]
+==== Install Minishift
+
+Make sure you have all prerequisites installed. Please check the list link:https://docs.okd.io/latest/minishift/getting-started/preparing-to-install.html[here].
+
+Download and put `minishift` in your $PATH by following steps link:https://docs.okd.io/latest/minishift/getting-started/installing.html#installing-manually[here].
+
+Verify installation by running following command, you should get version number.
+```bash
+minishift version
+```
+
+[[install-oc]]
+==== Install oc
+Please install and set up oc on your machine by visiting link:https://docs.okd.io/latest/cli_reference/get_started_cli.html#installing-the-cli[oc].
+
+Verify installation by running following command, you should get version number.
+```bash
+oc version
+```
+
+[[usage]]
+== Usage
+=== TL;DR
+```
+make minishift-start
+eval $(minishift docker-env)
+make dev-openshift
```

Comment:
Why not include everything in `make dev-openshift`? Why three commands?

Reply:
If we expose docker-deamon, then it can find locally built docker image sucessfully as we are using `imagePullPolicy: IfNotPresent` which is default in openshift. This way we'll avoid pushing locally built image to minishift registry and pulling it for each deploy which I believe will save lot of our time.

if we use `eval $(minishift docker-env)` inside any make target, all exported variables are only available to that make target's subprocess due to which minishift docker-daemon is not exposed to your host where we built cluster image locally using `docker build -t fabric8-services/fabric8-cluster:dev -f Dockerfile.dev .` To avoid we need to use three targets.